algopy_testing/src/algopy_testing/context.py

```diff
@@ -1,24 +1,495 @@
-import dataclasses
-from collections.abc import Generator
+from __future__ import annotations
+
+import random
 from contextlib import contextmanager
 from contextvars import ContextVar
+from typing import TYPE_CHECKING, Any, Generic, TypeVar, Unpack
+
+import algosdk
+
+from algopy_testing.constants import MAX_BYTES_SIZE, MAX_UINT64
+from algopy_testing.models.account import AccountData, AccountFields
+from algopy_testing.models.application import ApplicationFields
+from algopy_testing.models.asset import AssetFields
+from algopy_testing.models.global_state import GlobalFields
+from algopy_testing.models.itxn import ITxnFields
+from algopy_testing.models.txn import TxnFields
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+    import algopy
+    import algopy.gtxn
+
+    from algopy_testing.itxn import BaseInnerTransaction
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+class AlgopyTestContext(Generic[T]):
+    def __init__(self, contract_cls: type[T] | None = None) -> None:
+        self._contract_cls = (
+            contract_cls  # TODO: not utilized yet, just storing the reference for now
+        )
+        self.account_data: dict[str, AccountData] = {}
+        self.application_data: dict[int, ApplicationFields] = {}
+        self.asset_data: dict[int, AssetFields] = {}
+        self.inner_transactions: list[BaseInnerTransaction] = []
+        self.global_fields: dict[str, Any] = {}
+        self.txn_fields: dict[str, Any] = {}
+        self.itxn_fields: dict[str, Any] = {}
+        self.gtxns: list[algopy.gtxn.TransactionBase] = []
+        self.logs: list[str] = []
+        self._asset_id_count: int = 1
```

Comment:
Can use this trick to generate valid id's
```suggestion
        self._asset_id = iter(range(1001, 2**64))
```
Then when you need the next id it is just `next(self._asset_id)`